cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -0,0 +1,515 @@
+                .withUid(kafkaCr.getMetadata().getUid())
+                .withBlockOwnerDeletion(false)
+                .withController(false)
+                .build();
+        this.clusterCaConfig = kafkaCr.getSpec().getClusterCa();
+        this.clientsCaConfig = kafkaCr.getSpec().getClientsCa();
+        this.caLabels = Labels
+                .generateDefaultLabels(kafkaCr, Labels.APPLICATION_NAME, AbstractModel.STRIMZI_CLUSTER_OPERATOR_NAME)
+                .toMap();
+        this.clusterOperatorSecretLabels = Labels.fromResource(kafkaCr)
+                .withStrimziKind(reconciliation.kind())
+                .withStrimziCluster(reconciliation.name())
+                .withKubernetesName(Labels.APPLICATION_NAME)
+                .withKubernetesInstance(reconciliation.name())
+                .withKubernetesPartOf(reconciliation.name())
+                .withKubernetesManagedBy(AbstractModel.STRIMZI_CLUSTER_OPERATOR_NAME);
+        this.clusterCaCertLabels = clusterCaCertLabels(kafkaCr);
+        this.clusterCaCertAnnotations = clusterCaCertAnnotations(kafkaCr);
+    }
+
+    /**
+     * Utility method to extract the template labels from the Kafka CR.
+     *
+     * @param kafkaCr   Kafka CR
+     *
+     * @return  Map with the labels from the Kafka CR or empty map if the template is not set
+     */
+    private static Map<String, String> clusterCaCertLabels(Kafka kafkaCr)    {
+        if (kafkaCr.getSpec().getKafka() != null
+                && kafkaCr.getSpec().getKafka().getTemplate() != null
+                && kafkaCr.getSpec().getKafka().getTemplate().getClusterCaCert() != null
+                && kafkaCr.getSpec().getKafka().getTemplate().getClusterCaCert().getMetadata() != null
+                && kafkaCr.getSpec().getKafka().getTemplate().getClusterCaCert().getMetadata().getLabels() != null) {
+            return kafkaCr.getSpec().getKafka().getTemplate().getClusterCaCert().getMetadata().getLabels();
+        } else {
+            return Map.of();
+        }
+    }
+
+    /**
+     * Utility method to extract the template annotations from the Kafka CR.
+     *
+     * @param kafkaCr   Kafka CR
+     *
+     * @return  Map with the annotation from the Kafka CR or empty map if the template is not set
+     */
+    private static Map<String, String> clusterCaCertAnnotations(Kafka kafkaCr)    {
+        if (kafkaCr.getSpec().getKafka() != null
+                && kafkaCr.getSpec().getKafka().getTemplate() != null
+                && kafkaCr.getSpec().getKafka().getTemplate().getClusterCaCert() != null
+                && kafkaCr.getSpec().getKafka().getTemplate().getClusterCaCert().getMetadata() != null
+                && kafkaCr.getSpec().getKafka().getTemplate().getClusterCaCert().getMetadata().getAnnotations() != null) {
+            return kafkaCr.getSpec().getKafka().getTemplate().getClusterCaCert().getMetadata().getAnnotations();
+        } else {
+            return Map.of();
+        }
+    }
```

Comment:
Ditto as above, but for annotations?

Reply:
TBH, I think the `if` above is way more readable and easier to understand what it actually does.